roles/sriov/tasks/mlxconfig.yml

```diff
@@ -0,0 +1,77 @@
+---
+- name: Install MFT dependencies
+  ansible.builtin.yum:
+    name: "{{ sriov_os_pkgs }}"
+  become: true
+  when: sriov_install_enabled
+
+- name: Make working directory
+  ansible.builtin.file:
+    state: directory
+    path: "{{ sriov_workdir }}"
+    mode: "0700"
+  when: sriov_install_enabled
+
+- name: Download MFT
+  vars:
+    # Strip the file extension
+    no_ext: "{{ (sriov_mft_tarball | splitext)[0] }}"
+  ansible.builtin.unarchive:
+    src: "{{ sriov_mft_url }}"
+    dest: "{{ sriov_workdir }}"
+    remote_src: true
+    creates: "{{ sriov_workdir }}/{{ no_ext }}"
+  when: sriov_install_enabled
+
+- name: Determine if we can run mst_status
+  ansible.builtin.command: mst status
+  become: true
+  failed_when: false
+  changed_when: false
+  register: mst_status_result
+
+- name: Install MFT
+  ansible.builtin.command: "{{ sriov_workdir }}/{{ no_ext }}/install.sh"
+  vars:
+    # Strip the file extension
+    no_ext: "{{ (sriov_mft_tarball | splitext)[0] }}"
+  become: true
+  register: mft_install_result
+  changed_when: mft_install_result.rc == 0
```

Comment:
If the command fails then the task will fail, so this could just be `changed_when: true`

Reply:
fixed